src/dialogs/restart/dialog-restart.ts

```diff
@@ -0,0 +1,374 @@
+                  @request-selected=${this._restart}
+                >
+                  <div slot="graphic" class="icon-background restart">
+                    <ha-svg-icon .path=${mdiRefresh}></ha-svg-icon>
+                  </div>
+                  <span>
+                    ${this.hass.localize("ui.dialogs.restart.restart.title")}
+                  </span>
+                  <span slot="secondary">
+                    ${this.hass.localize(
+                      "ui.dialogs.restart.restart.description"
+                    )}
+                  </span>
+                </ha-list-item>
+                ${showRebootShutdown
+                  ? html`
+                      <div class="divider"></div>
+                      <ha-list-item
+                        graphic="avatar"
+                        twoline
+                        hasMeta
+                        @request-selected=${this._hostReboot}
+                      >
+                        <div slot="graphic" class="icon-background reboot">
+                          <ha-svg-icon .path=${mdiPowerCycle}></ha-svg-icon>
+                        </div>
+                        <span>
+                          ${this.hass.localize(
+                            "ui.dialogs.restart.reboot.title"
+                          )}
+                        </span>
+                        <span slot="secondary">
+                          ${this.hass.localize(
+                            "ui.dialogs.restart.reboot.description"
+                          )}
+                        </span>
+                      </ha-list-item>
+                      <ha-list-item
+                        graphic="avatar"
+                        twoline
+                        hasMeta
+                        @request-selected=${this._hostShutdown}
+                      >
+                        <div slot="graphic" class="icon-background shutdown">
+                          <ha-svg-icon .path=${mdiPower}></ha-svg-icon>
+                        </div>
+                        <span>
+                          ${this.hass.localize(
+                            "ui.dialogs.restart.shutdown.title"
+                          )}
+                        </span>
+                        <span slot="secondary">
+                          ${this.hass.localize(
+                            "ui.dialogs.restart.shutdown.confirm_action"
+                          )}
+                        </span>
+                      </ha-list-item>
+                    `
+                  : null}
+              </mwc-list>
+            `}
+      </ha-dialog>
+    `;
+  }
+
+  private async _reload(ev) {
+    if (!shouldHandleRequestSelectedEvent(ev)) {
+      return;
+    }
+    const reloadableDomains = componentsWithService(this.hass, "reload");
+
+    this.closeDialog();
+
+    showToast(this, {
+      message: this.hass.localize("ui.dialogs.restart.reload.reloading"),
+      duration: 2000,
```

Comment:
It now looks like the reloading is done when the toast disappears, but that is not true, I think a spinner would be better if we actually get that feedback from core.